frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -356,11 +358,22 @@ export const insightLogic = kea<insightLogicType>({
                 ...insight,
             }),
             setInsightMetadata: (state, { metadata }) => ({ ...state, ...metadata }),
-            [dashboardsModel.actionTypes.updateDashboardItem]: (state, { item }) => {
-                if (item.short_id === state.short_id) {
-                    return { ...item }
+            [dashboardsModel.actionTypes.updateDashboardItem]: (state, { item, dashboardIds }) => {
+                if (item.short_id !== state.short_id) {
+                    return state
+                }
+
+                const updatedInsight: Partial<InsightModel> = { ...item }
```

Comment:
Nit: I think you don't really need to clone this, if returning directly below?

Reply:
The insights are passed around so much it's probably safest to clone... but it can be simplified